scarlett/basics/talk.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: UTF-8 -*-
+
+# NOTE: Borrowed a great deal of knowledge from LISA project.
+
+import gobject
+gobject.threads_init()
+
+# Imports
+import scarlett
+import threading
+import os
+import gettext
+from Queue import Queue
+from time import sleep
+from subprocess import call
+# import urllib
+# from urllib import urlencode, urlopen
+from random import randint
+# from twisted.python import log
+from scarlett.basics import *
+from scarlett.constants import *
+import scarlett.basics.voice
+
+# System utterance definition : key : {(weight1, message1), (weight2, message2)}
+# Talk will randomly choose a message in list, balanced by weights
+_utterances = {
```

Comment:
@jalcine 

This is actually still a complete work in progress ... 

`where I want this to be:`

So i want `ScarlettTalk` to be a singleton class that you call whenever you need it ( Since it's using Gstreamer and I want to control which sound is played at any given point in time without them overlapping ). Right now a lot of this is boilerplate code for that, but it's actually example code from some other repos I've seen out there and using it as a base to get this working. 

`what it is currently:`

So the person using this code actually used pico2wav to take those base phrases, write them to to `/tmp` and play one of those responses whenever you speak to the application, hence the weights etc. I'm not a big fan of reading in data, and writing it to file every time a TTS is preformed ... so going to change it to work my way :)